src/modules/commander/failure_detector/FailureDetector.cpp

```diff
@@ -0,0 +1,108 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2018 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+* @file FailureDetector.cpp
+*
+* @author Mathieu Bresciani	<brescianimathieu@gmail.com>
+*
+*/
+
+#include "FailureDetector.hpp"
+
+FailureDetector::FailureDetector() :
+	ModuleParams(nullptr),
+	_sub_vehicle_attitude_setpoint(ORB_ID(vehicle_attitude_setpoint)),
+	_sub_vehicule_attitude(ORB_ID(vehicle_attitude))
+{
+}
+
+void
+FailureDetector::update_params()
+{
+	updateParams();
+}
+
+bool
+FailureDetector::update()
+{
+	bool updated(false);
+
+	updated = update_attitude_status();
```

Comment:
`return update_attitude_status();`

Reply:
I used that style to have the skeleton ready for other updates. For example, we'll add a check against altitude, the code will be:

``` c++
updated = update_attitude_status();
updated |= update_altitude_status();
updated |= update_other_status();

return updated;
```

I find it cleaner than

```c++
return  update_attitude_status() |  update_altitude_status() | update_other_status();
```